serviceregistration/consul/consul_service_registration.go

```diff
@@ -372,13 +387,25 @@ func (c *ConsulServiceRegistration) runEventDemuxer(waitGroup *sync.WaitGroup, s
 
 	for !shutdown {
 		select {
-		case <-c.notifyActiveCh:
+		case isActive := <-c.notifyActiveCh:
+			c.stateLock.Lock()
+			c.isActive = isActive
+			c.stateLock.Unlock()
```

Comment:
Optional: Instead of locking and unlocking consider using [uber/atomic.Bool](https://godoc.org/go.uber.org/atomic#Bool)

Reply:
Thanks! I think I will stick with this one because of how it's a `RWLock`. That will allow multiple concurrent reads, but lock when mutating the object. For that reason, I'd expect the reads to be faster using the `RWLock` than with an atomic bool.